Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceManagmentAndroidDeviceOwnerEnrollmentProfile/MSFT_IntuneDeviceManagmentAndroidDeviceOwnerEnrollmentProfile.psm1

```diff
@@ -0,0 +1,699 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter()]
+        [System.String]
+        $Id,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $DisplayName,
+
+        [Parameter()]
+        [System.String]
+        $AccountId,
+
+        [Parameter()]
+        [System.Boolean]
+        $ConfigureWifi,
+
+        [Parameter()]
+        [System.DateTime]
+        $CreatedDateTime,
+
+        [Parameter()]
+        [System.String]
+        $Description,
+
+        [Parameter()]
+        [System.Int32]
+        $EnrolledDeviceCount,
+
+        [Parameter()]
+        [System.String]
+        [ValidateSet( 'corporateOwnedDedicatedDevice', 'corporateOwnedFullyManaged', 'corporateOwnedWorkProfile', 'corporateOwnedAOSPUserlessDevice', 'corporateOwnedAOSPUserAssociatedDevice')]
+        $EnrollmentMode,
+
+        [Parameter()]
+        [ValidateSet( 'default', 'corporateOwnedDedicatedDeviceWithAzureADSharedMode', 'deviceStaging')]
+        $EnrollmentTokenType,
+
+        [Parameter()]
+        [System.Int32]
+        $EnrollmentTokenUsageCount,
+
+        [Parameter()]
+        [System.Boolean]
+        $IsTeamsDeviceProfile,
+
+        [Parameter()]
+        [System.DateTime]
+        $LastModifiedDateTime,
+
+        [Parameter()]
+        [System.String]
+        $QrCodeContent,
+
+        [Parameter()]
+        [Microsoft.Graph.IMicrosoftGraphMimeContent]
+        $QrCodeImage,
+
+        [Parameter()]
+        [System.String[]]
+        $RoleScopeTagIds,
+
+        [Parameter()]
+        [System.DateTime]
+        $TokenCreationDateTime,
+
+        [Parameter()]
+        [System.DateTime]
+        $TokenExpirationDateTime,
+
+        [Parameter()]
+        [System.String]
+        $TokenValue,
+
+        [Parameter()]
+        [System.Boolean]
+        $WifiHidden,
+
+        [Parameter()]
+        [System.Security.SecureString]
+        $WifiPassword,
```

Comment:
Secure String is not a valid MOF type. Instead, we need to define as PSCredential, where the username part can be anything, and the password piece is where you set the secure string value. You can refer to this example:
https://github.com/microsoft/Microsoft365DSC/blob/Dev/Modules/Microsoft365DSC/DSCResources/MSFT_AADUser/MSFT_AADUser.psm1#L37